src/compiler/checker.ts

```diff
@@ -16406,9 +16406,13 @@ namespace ts {
                 }
             }
             const suggestion = getSuggestionForNonexistentProperty(propNode, containingType);
+            const promisedType = getPromisedTypeOfPromise(containingType);
             if (suggestion !== undefined) {
                 errorInfo = chainDiagnosticMessages(errorInfo, Diagnostics.Property_0_does_not_exist_on_type_1_Did_you_mean_2, declarationNameToString(propNode), typeToString(containingType), suggestion);
             }
+            else if (promisedType && getPropertyOfType(promisedType, propNode.escapedText)) {

```

Comment:
nit, consider moving `const promisedType = getPromisedTypeOfPromise(containingType);` inside an `else` block for `if (suggestion !== undefined)`. it just makes sure we only compute it if we need it.